src/LayoutManager/LayoutStrategies/LayoutStrategy.ts

```diff
@@ -0,0 +1,186 @@
+import { getObjectBounds } from './utils';
+
+export abstract class LayoutStrategy {
+  abstract calcLayoutResult(
+    context: StrictLayoutContext,
+    objects: FabricObject[]
+  ): LayoutStrategyResult | undefined;
+
+  /**
+   * Override this method to customize layout.
+   * A wrapper around {@link getObjectsBoundingBox}
+   */
+  calcBoundingBox(
+    objects: FabricObject[],
+    context: StrictLayoutContext
+  ): LayoutStrategyResult | undefined {
+    if (context.type === 'initialization') {
+      return this.calcInitialBoundingBox(objects, context);
+    } else if (context.type === 'imperative' && context.overrides) {
+      return context.overrides;
+    } else {
+      return this.getObjectsBoundingBox(context.target, objects);
+    }
+  }
+
+  shouldLayoutClipPath() {
+    return true;
+  }
+
+  /**
+   * Calculate the bbox of objects (if necessary) and translate it according to options received from the constructor (left, top, width, height, originX, originY)
+   * so it is placed in the center of the bbox received from the constructor
+   *
+   * @todo revisit stroke calculations and respect angle and skew so that we can set them before initial layout
+   */
+  protected calcInitialBoundingBox(
+    objects: FabricObject[],
+    context: StrictLayoutContext
+  ): LayoutStrategyResult | undefined {
+    const { target } = context;
+    const options = context.options || {},
+      hasX = typeof options.left === 'number',
+      hasY = typeof options.top === 'number',
+      hasWidth = typeof options.width === 'number',
+      hasHeight = typeof options.height === 'number';
+
+    //  performance enhancement
+    //  skip layout calculation if bbox is defined
+    if (
+      (hasX &&
+        hasY &&
+        hasWidth &&
+        hasHeight &&
+        context.objectsRelativeToGroup) ||
+      objects.length === 0
+    ) {
+      //  return nothing to skip layout
+      return;
+    }
+
+    const bbox =
+      this.getObjectsBoundingBox(target, objects) ||
+      ({} as LayoutStrategyResult);
+    const { centerX = 0, centerY = 0, width: w = 0, height: h = 0 } = bbox;
+    const {
+      left: x,
+      top: y,
+      width: initialWidth,
+      height: initialHeight,
+      originX,
+      originY,
+    } = target;
+    const width = hasWidth ? initialWidth : w,
+      height = hasHeight ? initialHeight : h,
+      calculatedCenter = new Point(centerX, centerY),
+      origin = new Point(resolveOrigin(originX), resolveOrigin(originY)),
+      size = new Point(width, height),
+      strokeWidthVector = target._getTransformedDimensions({
+        width: 0,
+        height: 0,
+      }),
+      sizeAfter = target._getTransformedDimensions({
+        width: width,
+        height: height,
+        strokeWidth: 0,
+      }),
+      bboxSizeAfter = target._getTransformedDimensions({
+        width: bbox.width,
+        height: bbox.height,
+        strokeWidth: 0,
+      });
+
+    //  calculate center and correction
```

Comment:
What's this correction? It looks like you want to handle when the group has initial position specified, but the objects are absolute so you calculate a correction for the objects so that they end where the group position is? It look however a use case whose handling is leaking the implementation detail `correction`, which makes the rest more complicated as well.

Reply:
The correction value is the nastiest bit of logic of the layout manager.
It is needed because objects are relative to the group center but then the group changes size so an object that is expected to align to the top left corner of group needs to apply a correction to its position so it relates to the new center - I will add this to the docs

Reply:
I have an elaborate solution for dropping the need for the correction value... but it might degrade perf and limit customization.
I might expose a layout strategy that does that for the common good